### DISCUSSION TITLE
Download speed slow behind nginx

### DISCUSSION THREAD

**Discussion by yurixahri** [rejetto/hfs#636] [Q&A]
I was testing my internet speed after doing some RAID setup. The local transfer speed is 2.5Gbps, Lan speed is 1Gbps, but downloading  files behind nginx only get around ~ 22MB/s. I tried to down via local ip and the speed got back to ~ 100MB/s. I'm not familiar with reverse proxy that much, can you help me solve this problem? 
![image](https://github.com/rejetto/hfs/assets/111081318/6f474386-135b-41e8-a002-101b63f35c63)

**Comment by rejetto** [comment]
i doubt nginx is the problem, as it's known to be fast.
something else must be affecting you "after it".
but this is not an hfs issue, so i'm moving you to "discussions".

**Reply by yurixahri** [reply]
i'm trying another reverse proxy like Caddy or Ngrok to see if this getting any better. nginx on windows kinda sloppy and not as good as on linux.

**Reply by yurixahri** [reply]
So the problem is not because of reverse proxy, i tested files transfer both in local ip and public ip, the local ip is very fast (~100MB/s) but the public ip is slow (20MB/s). My internet plan is 1gbps up/down, i borrowed a friend to test the download speed via domain and surprisingly, his download speed is higher than me (~50MB/s), and me, who sit next to the server have slower speed than him. I'm really confused right now, after all the networking.

**Comment by rejetto** [comment]
so, you tested public ip without nginx (directly hfs), right?

**Reply by yurixahri** [reply]
yes

**Comment by rejetto** [comment]
i see.
i'm no network expert, maybe someone else...
